# 1. 라이브러리 임포트

In [1]:
import pandas as pd
import numpy as np
from PIL import Image
import matplotlib.pyplot as plt
import json

import requests
import glob
import os

import urllib.request
from PIL import Image
import time
from tqdm import tqdm

# 2. 데이터 로드

In [2]:
df = pd.read_csv("Total_Data_1.csv")

# 3. OCR을 적용된 결과 JSON 파일로 만들기

In [9]:
df.reset_index(inplace=True)

* 쿠팡 파트너스에 대해서

In [10]:
df1 = df[df["광고 분류2"]==1].reset_index(drop=True)
df1.head(1)

index                                           url  \
0      9  https://blog.naver.com/anfwjdtp/222565268273   

                            title  post_date      검색어 브랜드 명  \
0  가정용 전자동 커피머신★빈프레소 강제(?) 사용 후기ㅋ   20211111  빈프레소 리뷰  빈프레소   

                                             content  content_cnt  \
0  #가정용전자동커피머신 #가정용에스프레소머신 #가정용원두커피머신 #전자동커피머신 #빈...         1870   

   content_line                               quote  ...  ♡ 키워드 ♥ 키워드  \
0            17  ['가정용 전자동 커피머신★빈프레소 강제(?) 사용 후기 ']  ...      0     0   

  구매후기 키워드  한달 키워드  두달 키워드  세달 키워드  WEB_광고 분류1  WEB_광고 분류2  광고 분류1  광고 분류2  
0        0       0       0       0           1           1       1       1  

[1 rows x 51 columns]

* 협찬, 체험단 등 소정료를 받아 작성된 광고성 블로그 리뷰

In [11]:
df2 = df[df["광고 분류2"]==2].reset_index(drop=True)
df2.head(1)

index                                                url  \
0      0  https://blog.naver.com/PostView.naver?blogId=e...   

                      title  post_date      검색어 브랜드 명  \
0  무선 무릎찜질기 메이크맨 무르비2 사용 후기   20220124  무르비2 리뷰  메이크맨   

                                             content  content_cnt  \
0  저는 여름보다 유달리 겨울 추위를 잘 타는 체질이라 ... 히터가 작동되는 사무실에...         2435   

   content_line                                     quote  ...  ♡ 키워드 ♥ 키워드  \
0            22  ['편리한 무선 온열찜질기진동 마사지 기능옥이 들어있어 원적외선 효과']  ...      0     0   

  구매후기 키워드  한달 키워드  두달 키워드  세달 키워드  WEB_광고 분류1  WEB_광고 분류2  광고 분류1  광고 분류2  
0        0       0       0       0           0           0       1       2  

[1 rows x 51 columns]

* 소정료를 받지 않고 직접 구매하여 작성된 솔직한 블로그 리뷰

In [12]:
df0 = df[df["광고 분류2"]==0].reset_index(drop=True)
df0.head(1)

index                                                url  \
0      7  https://blog.naver.com/PostView.naver?blogId=i...   

                        title  post_date      검색어 브랜드 명  \
0  빈프레소 커피머신, 가정용 전자동 커피머신 후기   20220304  빈프레소 리뷰  빈프레소   

                                             content  content_cnt  \
0  남편이 우리집 다음 커피머신기는 빈프레소라면서 빈프레소가 요즘 핫하다면서 그런데 비...         2329   

   content_line          quote  ...  ♡ 키워드 ♥ 키워드 구매후기 키워드  한달 키워드  두달 키워드  \
0            25  ['빈프레소 커피머신']  ...      1     0        0       0       0   

   세달 키워드  WEB_광고 분류1  WEB_광고 분류2  광고 분류1  광고 분류2  
0       0           0           0       0       0  

[1 rows x 51 columns]

## 3.1 저장할 데이터 폴더 구축

In [13]:
import os
from tqdm import tqdm

os.makedirs("Images",exist_ok=True)
os.makedirs("Images/0",exist_ok=True)
os.makedirs("Images/1",exist_ok=True)
os.makedirs("Images/2",exist_ok=True)

os.makedirs("annotation",exist_ok=True)
os.makedirs("annotation/0",exist_ok=True)
os.makedirs("annotation/1",exist_ok=True)
os.makedirs("annotation/2",exist_ok=True)

In [15]:
df1.shape[0], df2.shape[0],df0.shape[0]

(172, 373, 1980)

* 쿠팡 파트너스 : 172
* 협찬을 받아 작성된 글 : 373  
* 협찬을 받지 않고 작성된 글 : 1980

## 3.2 json 파일로 저장

### 3.2.1 1번 -> 쿠팡 파트너스는 일정 수수료 받는다 (광고)

In [16]:
for i in tqdm(range(df1.shape[0])):
    if len(str(df1.iloc[i]["last_img"])) !=3: 
        try:
            opener = urllib.request.build_opener()
            opener.addheaders = [('User-Agent','Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/36.0.1941.0 Safari/537.36')]
            urllib.request.install_opener(opener)
            
            save_number = df1.iloc[i]['index']
            save_img_path = f"./Images/1/{save_number}_class1.png"
            save_json_path = f"./annotation/1/{save_number}_class1.json"
            
            # 이미지 저장.
            urllib.request.urlretrieve(df1.iloc[i]["last_img"], save_img_path)
            
            ### 이미지 글자 추출
            time.sleep(3)
            files = {"file": open(save_img_path, "rb").read()}
            response = requests.post('http://27.255.77.102:5000/evaluation',files=files)
            if response.status_code == 200:
                feature_text = response.json()
                if len(feature_text) ==0:
                    print(save_number,"내용이 없다고? !!")
                    print(feature_text)
                    print()
                    continue
                else:
                    json_files = {"file": open(save_img_path, "rb").read()}
                    with open(save_json_path,'w',  encoding='utf-8') as f:
                        json.dump(feature_text, f, ensure_ascii=False, indent='\t')
        except:
            pass

  2%|█▉                                                                                | 4/172 [00:15<10:49,  3.87s/it]

36 내용이 없다고? !!
{}



  3%|██▊                                                                               | 6/172 [00:23<10:46,  3.90s/it]

48 내용이 없다고? !!
{}



 42%|█████████████████████████████████▉                                               | 72/172 [04:27<06:25,  3.85s/it]

742 내용이 없다고? !!
{}



 43%|██████████████████████████████████▊                                              | 74/172 [04:36<06:44,  4.13s/it]

755 내용이 없다고? !!
{}



 49%|███████████████████████████████████████▌                                         | 84/172 [05:18<06:05,  4.15s/it]

1084 내용이 없다고? !!
{}



 53%|██████████████████████████████████████████▊                                      | 91/172 [05:44<05:40,  4.20s/it]

1509 내용이 없다고? !!
{}



 64%|███████████████████████████████████████████████████▏                            | 110/172 [06:53<03:00,  2.92s/it]

1754 내용이 없다고? !!
{}



 74%|███████████████████████████████████████████████████████████▌                    | 128/172 [08:06<02:55,  3.99s/it]

1968 내용이 없다고? !!
{}



 78%|██████████████████████████████████████████████████████████████▊                 | 135/172 [08:36<02:36,  4.24s/it]

2043 내용이 없다고? !!
{}



 90%|███████████████████████████████████████████████████████████████████████▋        | 154/172 [09:54<01:14,  4.12s/it]

2392 내용이 없다고? !!
{}



 91%|████████████████████████████████████████████████████████████████████████▌       | 156/172 [10:02<01:05,  4.06s/it]

2394 내용이 없다고? !!
{}



 98%|██████████████████████████████████████████████████████████████████████████████▌ | 169/172 [10:53<00:12,  4.10s/it]

2460 내용이 없다고? !!
{}



100%|████████████████████████████████████████████████████████████████████████████████| 172/172 [11:05<00:00,  3.87s/it]


### 3.2.2 2번 -> 체험단, 협찬 , 소정료 받아서 씀(광고)

In [17]:
for i in tqdm(range(df2.shape[0])):
    if len(str(df2.iloc[i]["last_img"])) !=3: 
        try:
            opener = urllib.request.build_opener()
            opener.addheaders = [('User-Agent','Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/36.0.1941.0 Safari/537.36')]
            urllib.request.install_opener(opener)
            
            save_number = df2.iloc[i]['index']
            save_img_path = f"Images/2/{save_number}_class2.png"
            save_json_path = f"annotation/2/{save_number}_class2.json"
            
            # 이미지 저장.
            urllib.request.urlretrieve(df2.iloc[i]["last_img"], save_img_path)
            
            ### 이미지 글자 추출
            time.sleep(1)
            files = {"file": open(save_img_path, "rb").read()}
            response = requests.post('http://27.255.77.102:5000/evaluation',files=files)
            if response.status_code == 200:
                feature_text = response.json()
                if len(feature_text) ==0:
                    print(save_number,"내용이 없다고? !!")
                    print(feature_text)
                    print()
                    continue
                else:
                    json_files = {"file": open(save_img_path, "rb").read()}
                    with open(save_json_path,'w',  encoding='utf-8') as f:
                        json.dump(feature_text, f, ensure_ascii=False, indent='\t')
        except:
            pass

  1%|▋                                                                                 | 3/373 [00:09<17:58,  2.91s/it]

2 내용이 없다고? !!
{}



  1%|▉                                                                                 | 4/373 [00:11<14:37,  2.38s/it]

3 내용이 없다고? !!
{}



  4%|███▎                                                                             | 15/373 [00:40<19:32,  3.28s/it]

53 내용이 없다고? !!
{}



  6%|████▉                                                                            | 23/373 [01:01<16:39,  2.85s/it]

172 내용이 없다고? !!
{}



  8%|██████▎                                                                          | 29/373 [01:14<12:09,  2.12s/it]

183 내용이 없다고? !!
{}



 13%|██████████▊                                                                      | 50/373 [02:00<10:09,  1.89s/it]

299 내용이 없다고? !!
{}



 14%|███████████▎                                                                     | 52/373 [02:04<09:14,  1.73s/it]

342 내용이 없다고? !!
{}



 20%|████████████████▎                                                                | 75/373 [03:00<10:32,  2.12s/it]

691 내용이 없다고? !!
{}



 23%|██████████████████▍                                                              | 85/373 [03:22<08:03,  1.68s/it]

711 내용이 없다고? !!
{}



 25%|████████████████████▍                                                            | 94/373 [03:39<08:25,  1.81s/it]

725 내용이 없다고? !!
{}



 27%|█████████████████████▍                                                           | 99/373 [03:50<09:45,  2.14s/it]

750 내용이 없다고? !!
{}



 31%|█████████████████████████                                                       | 117/373 [04:32<09:53,  2.32s/it]

916 내용이 없다고? !!
{}



 35%|████████████████████████████▎                                                   | 132/373 [05:03<08:27,  2.11s/it]

958 내용이 없다고? !!
{}



 36%|████████████████████████████▌                                                   | 133/373 [05:06<09:13,  2.30s/it]

963 내용이 없다고? !!
{}



 38%|██████████████████████████████                                                  | 140/373 [05:23<09:19,  2.40s/it]

1067 내용이 없다고? !!
{}



 43%|██████████████████████████████████▎                                             | 160/373 [06:07<08:30,  2.40s/it]

1476 내용이 없다고? !!
{}



 43%|██████████████████████████████████▌                                             | 161/373 [06:09<08:12,  2.33s/it]

1477 내용이 없다고? !!
{}



 46%|████████████████████████████████████▍                                           | 170/373 [06:28<06:56,  2.05s/it]

1488 내용이 없다고? !!
{}



 49%|███████████████████████████████████████▍                                        | 184/373 [07:01<08:40,  2.75s/it]

1504 내용이 없다고? !!
{}



 50%|████████████████████████████████████████▎                                       | 188/373 [07:10<07:48,  2.53s/it]

1528 내용이 없다고? !!
{}



 51%|████████████████████████████████████████▊                                       | 190/373 [07:14<06:49,  2.24s/it]

1530 내용이 없다고? !!
{}



 52%|█████████████████████████████████████████▍                                      | 193/373 [07:20<06:22,  2.13s/it]

1533 내용이 없다고? !!
{}



 53%|██████████████████████████████████████████                                      | 196/373 [07:27<06:51,  2.32s/it]

1536 내용이 없다고? !!
{}



 53%|██████████████████████████████████████████▋                                     | 199/373 [07:34<06:28,  2.23s/it]

1544 내용이 없다고? !!
{}



 54%|███████████████████████████████████████████▎                                    | 202/373 [07:42<07:04,  2.48s/it]

1552 내용이 없다고? !!
{}



 55%|████████████████████████████████████████████▍                                   | 207/373 [07:51<05:34,  2.01s/it]

1566 내용이 없다고? !!
{}



 59%|███████████████████████████████████████████████▍                                | 221/373 [08:17<04:23,  1.73s/it]

1617 내용이 없다고? !!
{}



 60%|███████████████████████████████████████████████▌                                | 222/373 [08:18<04:18,  1.71s/it]

1618 내용이 없다고? !!
{}



 62%|█████████████████████████████████████████████████▎                              | 230/373 [08:33<04:17,  1.80s/it]

1661 내용이 없다고? !!
{}



 62%|█████████████████████████████████████████████████▉                              | 233/373 [08:39<04:34,  1.96s/it]

1700 내용이 없다고? !!
{}



 63%|██████████████████████████████████████████████████▏                             | 234/373 [08:41<04:14,  1.83s/it]

1759 내용이 없다고? !!
{}



 64%|███████████████████████████████████████████████████                             | 238/373 [08:48<04:07,  1.83s/it]

1765 내용이 없다고? !!
{}



 65%|████████████████████████████████████████████████████▎                           | 244/373 [09:02<04:37,  2.15s/it]

1773 내용이 없다고? !!
{}



 70%|███████████████████████████████████████████████████████▉                        | 261/373 [09:36<03:59,  2.14s/it]

1810 내용이 없다고? !!
{}



 72%|█████████████████████████████████████████████████████████▍                      | 268/373 [09:51<03:43,  2.13s/it]

1833 내용이 없다고? !!
{}



 75%|████████████████████████████████████████████████████████████▎                   | 281/373 [10:15<03:10,  2.07s/it]

1859 내용이 없다고? !!
{}



 79%|███████████████████████████████████████████████████████████████▍                | 296/373 [10:41<02:24,  1.88s/it]

1971 내용이 없다고? !!
{}



 82%|█████████████████████████████████████████████████████████████████▏              | 304/373 [11:01<03:25,  2.97s/it]

2157 내용이 없다고? !!
{}



 90%|████████████████████████████████████████████████████████████████████████        | 336/373 [12:06<01:09,  1.87s/it]

2377 내용이 없다고? !!
{}



 91%|████████████████████████████████████████████████████████████████████████▋       | 339/373 [12:12<01:07,  1.99s/it]

2385 내용이 없다고? !!
{}



 92%|█████████████████████████████████████████████████████████████████████████▊      | 344/373 [12:22<00:59,  2.04s/it]

2422 내용이 없다고? !!
{}



100%|████████████████████████████████████████████████████████████████████████████████| 373/373 [13:20<00:00,  2.15s/it]


### 3.2.3 일반 사용자들이 수수료나 협찬 등 제외하고 직접 돈 주고 쓴 후기들 (진솔한 리뷰)

In [18]:
import urllib.request
from PIL import Image
import time
from tqdm import tqdm

for i in tqdm(range(df0.shape[0])):
    if len(str(df0.iloc[i]["last_img"])) !=3: 
        try:
            save_number = df0.iloc[i]['index']
            save_img_path = f"Images/0/{save_number}_class0.png"
            save_json_path = f"annotation/0/{save_number}_class0.json"
            
            # 이미지 저장.
            urllib.request.urlretrieve(df0.iloc[i]["last_img"], save_img_path)
            
            ### 이미지 글자 추출
            time.sleep(1)
            files = {"file": open(save_img_path, "rb").read()}
            response = requests.post('http://27.255.77.102:5000/evaluation',files=files)
            if response.status_code == 200:
                feature_text = response.json()
                if len(feature_text) ==0:
                    print(save_number,"내용이 없다고? !!")
                    print(feature_text)
                    print()
                    continue
                else:
                    json_files = {"file": open(save_img_path, "rb").read()}
                    with open(save_json_path,'w',  encoding='utf-8') as f:
                        json.dump(feature_text, f, ensure_ascii=False, indent='\t')
        except:
            pass

  0%|▏                                                                              | 5/1980 [00:10<1:08:28,  2.08s/it]

15 내용이 없다고? !!
{}



  1%|▋                                                                             | 18/1980 [00:38<1:02:13,  1.90s/it]

34 내용이 없다고? !!
{}



  1%|▉                                                                               | 23/1980 [00:47<56:59,  1.75s/it]

40 내용이 없다고? !!
{}



  2%|█▌                                                                              | 38/1980 [01:15<59:15,  1.83s/it]

58 내용이 없다고? !!
{}



  2%|█▋                                                                            | 42/1980 [01:23<1:02:05,  1.92s/it]

62 내용이 없다고? !!
{}



  2%|█▊                                                                            | 47/1980 [01:32<1:01:37,  1.91s/it]

67 내용이 없다고? !!
{}



  3%|██▎                                                                           | 58/1980 [01:57<1:10:14,  2.19s/it]

78 내용이 없다고? !!
{}



  3%|██▎                                                                           | 59/1980 [01:58<1:04:47,  2.02s/it]

79 내용이 없다고? !!
{}



  3%|██▎                                                                           | 60/1980 [02:00<1:06:13,  2.07s/it]

80 내용이 없다고? !!
{}



  3%|██▌                                                                           | 64/1980 [02:08<1:03:02,  1.97s/it]

84 내용이 없다고? !!
{}



  3%|██▌                                                                           | 65/1980 [02:10<1:04:28,  2.02s/it]

85 내용이 없다고? !!
{}



  3%|██▋                                                                             | 67/1980 [02:14<58:51,  1.85s/it]

87 내용이 없다고? !!
{}



  4%|██▉                                                                             | 73/1980 [02:25<59:56,  1.89s/it]

93 내용이 없다고? !!
{}



  4%|██▉                                                                           | 74/1980 [02:28<1:07:03,  2.11s/it]

94 내용이 없다고? !!
{}



  4%|███▏                                                                          | 80/1980 [02:42<1:14:29,  2.35s/it]

101 내용이 없다고? !!
{}



  4%|███▏                                                                          | 82/1980 [02:46<1:15:59,  2.40s/it]

104 내용이 없다고? !!
{}



  4%|███▎                                                                          | 84/1980 [02:50<1:04:34,  2.04s/it]

106 내용이 없다고? !!
{}



  5%|███▉                                                                         | 102/1980 [03:27<1:18:01,  2.49s/it]

127 내용이 없다고? !!
{}



  5%|████                                                                         | 106/1980 [03:36<1:09:00,  2.21s/it]

132 내용이 없다고? !!
{}



  6%|████▍                                                                          | 112/1980 [03:47<58:53,  1.89s/it]

140 내용이 없다고? !!
{}



  6%|████▋                                                                          | 119/1980 [04:00<57:35,  1.86s/it]

147 내용이 없다고? !!
{}



  6%|████▉                                                                        | 126/1980 [04:16<1:07:17,  2.18s/it]

154 내용이 없다고? !!
{}



  7%|█████                                                                        | 131/1980 [04:27<1:09:20,  2.25s/it]

160 내용이 없다고? !!
{}



  7%|█████▍                                                                       | 140/1980 [04:45<1:02:14,  2.03s/it]

169 내용이 없다고? !!
{}



  7%|█████▍                                                                       | 141/1980 [04:48<1:08:38,  2.24s/it]

170 내용이 없다고? !!
{}



  7%|█████▌                                                                       | 142/1980 [04:50<1:09:20,  2.26s/it]

171 내용이 없다고? !!
{}



  7%|█████▋                                                                       | 146/1980 [04:58<1:01:01,  2.00s/it]

176 내용이 없다고? !!
{}



  8%|█████▉                                                                       | 152/1980 [05:11<1:10:27,  2.31s/it]

191 내용이 없다고? !!
{}



  8%|██████                                                                       | 156/1980 [05:19<1:02:11,  2.05s/it]

211 내용이 없다고? !!
{}



  8%|██████▎                                                                        | 158/1980 [05:23<59:37,  1.96s/it]

213 내용이 없다고? !!
{}



  8%|██████▍                                                                        | 160/1980 [05:26<58:20,  1.92s/it]

216 내용이 없다고? !!
{}



  8%|██████▍                                                                      | 166/1980 [05:42<1:12:30,  2.40s/it]

223 내용이 없다고? !!
{}



  9%|██████▋                                                                      | 171/1980 [05:53<1:05:07,  2.16s/it]

228 내용이 없다고? !!
{}



  9%|███████                                                                      | 182/1980 [06:17<1:09:59,  2.34s/it]

239 내용이 없다고? !!
{}



  9%|███████▏                                                                     | 184/1980 [06:21<1:03:22,  2.12s/it]

241 내용이 없다고? !!
{}



  9%|███████▍                                                                       | 187/1980 [06:27<57:23,  1.92s/it]

247 내용이 없다고? !!
{}



 10%|███████▋                                                                     | 199/1980 [06:53<1:00:33,  2.04s/it]

261 내용이 없다고? !!
{}



 10%|████████                                                                       | 203/1980 [07:01<57:05,  1.93s/it]

265 내용이 없다고? !!
{}



 10%|████████                                                                     | 206/1980 [07:07<1:02:37,  2.12s/it]

268 내용이 없다고? !!
{}



 11%|████████▏                                                                    | 209/1980 [07:17<1:19:31,  2.69s/it]

272 내용이 없다고? !!
{}



 11%|████████▌                                                                    | 221/1980 [07:47<1:12:11,  2.46s/it]

284 내용이 없다고? !!
{}



 11%|████████▉                                                                      | 225/1980 [07:54<56:28,  1.93s/it]

288 내용이 없다고? !!
{}



 11%|█████████                                                                      | 227/1980 [07:58<54:14,  1.86s/it]

290 내용이 없다고? !!
{}



 12%|█████████                                                                      | 228/1980 [08:00<56:20,  1.93s/it]

291 내용이 없다고? !!
{}



 12%|█████████▎                                                                     | 234/1980 [08:11<53:02,  1.82s/it]

300 내용이 없다고? !!
{}



 12%|█████████▍                                                                     | 235/1980 [08:14<58:10,  2.00s/it]

301 내용이 없다고? !!
{}



 13%|█████████▊                                                                   | 253/1980 [08:46<1:01:57,  2.15s/it]

324 내용이 없다고? !!
{}



 13%|█████████▉                                                                   | 254/1980 [08:48<1:02:55,  2.19s/it]

325 내용이 없다고? !!
{}



 13%|██████████▎                                                                    | 259/1980 [08:57<54:37,  1.90s/it]

334 내용이 없다고? !!
{}



 13%|██████████▎                                                                    | 260/1980 [08:59<54:50,  1.91s/it]

340 내용이 없다고? !!
{}



 14%|███████████                                                                    | 276/1980 [09:29<48:12,  1.70s/it]

363 내용이 없다고? !!
{}



 14%|███████████                                                                    | 278/1980 [09:32<47:42,  1.68s/it]

365 내용이 없다고? !!
{}



 14%|██████████▉                                                                  | 281/1980 [09:39<1:00:57,  2.15s/it]

368 내용이 없다고? !!
{}



 14%|███████████▎                                                                   | 282/1980 [09:41<55:57,  1.98s/it]

369 내용이 없다고? !!
{}



 15%|███████████▌                                                                 | 297/1980 [10:15<1:07:10,  2.39s/it]

385 내용이 없다고? !!
{}



 15%|███████████▌                                                                 | 298/1980 [10:17<1:00:58,  2.18s/it]

386 내용이 없다고? !!
{}



 15%|████████████                                                                   | 303/1980 [10:28<55:07,  1.97s/it]

393 내용이 없다고? !!
{}



 15%|████████████▏                                                                  | 306/1980 [10:33<54:53,  1.97s/it]

396 내용이 없다고? !!
{}



 16%|████████████▉                                                                  | 323/1980 [11:08<55:14,  2.00s/it]

416 내용이 없다고? !!
{}



 16%|████████████▉                                                                  | 325/1980 [11:11<49:38,  1.80s/it]

419 내용이 없다고? !!
{}



 17%|█████████████                                                                  | 328/1980 [11:16<47:56,  1.74s/it]

422 내용이 없다고? !!
{}



 17%|█████████████▎                                                                 | 334/1980 [11:27<49:59,  1.82s/it]

428 내용이 없다고? !!
{}



 17%|█████████████▎                                                                 | 335/1980 [11:30<53:25,  1.95s/it]

429 내용이 없다고? !!
{}



 18%|█████████████▊                                                                 | 347/1980 [12:00<58:52,  2.16s/it]

442 내용이 없다고? !!
{}



 18%|██████████████▎                                                                | 360/1980 [12:27<58:24,  2.16s/it]

455 내용이 없다고? !!
{}



 19%|██████████████▊                                                                | 371/1980 [12:49<49:52,  1.86s/it]

466 내용이 없다고? !!
{}



 19%|██████████████▊                                                                | 372/1980 [12:51<47:39,  1.78s/it]

467 내용이 없다고? !!
{}



 19%|██████████████▉                                                                | 373/1980 [12:53<49:00,  1.83s/it]

468 내용이 없다고? !!
{}



 19%|███████████████▏                                                               | 380/1980 [13:05<46:32,  1.75s/it]

475 내용이 없다고? !!
{}



 20%|███████████████▍                                                               | 387/1980 [13:19<53:42,  2.02s/it]

482 내용이 없다고? !!
{}



 20%|███████████████▍                                                               | 388/1980 [13:21<50:32,  1.90s/it]

483 내용이 없다고? !!
{}



 20%|███████████████▌                                                               | 389/1980 [13:23<49:56,  1.88s/it]

484 내용이 없다고? !!
{}



 20%|███████████████▋                                                               | 392/1980 [13:29<54:49,  2.07s/it]

487 내용이 없다고? !!
{}



 20%|███████████████▊                                                               | 395/1980 [13:35<47:52,  1.81s/it]

491 내용이 없다고? !!
{}



 20%|███████████████▊                                                               | 396/1980 [13:36<47:10,  1.79s/it]

492 내용이 없다고? !!
{}



 20%|███████████████▊                                                               | 397/1980 [13:38<49:10,  1.86s/it]

493 내용이 없다고? !!
{}



 20%|███████████████▉                                                               | 401/1980 [13:48<52:36,  2.00s/it]

498 내용이 없다고? !!
{}



 20%|████████████████                                                               | 402/1980 [13:50<51:44,  1.97s/it]

499 내용이 없다고? !!
{}



 20%|████████████████▏                                                              | 405/1980 [13:56<53:57,  2.06s/it]

502 내용이 없다고? !!
{}



 21%|████████████████▍                                                              | 413/1980 [14:11<44:25,  1.70s/it]

510 내용이 없다고? !!
{}



 21%|████████████████▌                                                              | 416/1980 [14:13<28:09,  1.08s/it]

513 내용이 없다고? !!
{}



 21%|████████████████▋                                                              | 417/1980 [14:15<33:47,  1.30s/it]

515 내용이 없다고? !!
{}



 21%|████████████████▍                                                            | 422/1980 [14:28<1:05:26,  2.52s/it]

520 내용이 없다고? !!
{}



 21%|████████████████▉                                                              | 423/1980 [14:30<59:22,  2.29s/it]

521 내용이 없다고? !!
{}



 22%|█████████████████▏                                                             | 430/1980 [14:39<34:24,  1.33s/it]

528 내용이 없다고? !!
{}



 22%|█████████████████▍                                                             | 437/1980 [14:55<56:45,  2.21s/it]

537 내용이 없다고? !!
{}



 22%|█████████████████▏                                                           | 442/1980 [15:08<1:09:16,  2.70s/it]

543 내용이 없다고? !!
{}



 22%|█████████████████▊                                                             | 445/1980 [15:13<51:45,  2.02s/it]

546 내용이 없다고? !!
{}



 23%|█████████████████▊                                                             | 447/1980 [15:18<52:14,  2.04s/it]

549 내용이 없다고? !!
{}



 23%|█████████████████▊                                                             | 448/1980 [15:20<52:38,  2.06s/it]

550 내용이 없다고? !!
{}



 23%|██████████████████▏                                                            | 455/1980 [15:33<47:48,  1.88s/it]

558 내용이 없다고? !!
{}



 23%|██████████████████▏                                                            | 456/1980 [15:35<46:59,  1.85s/it]

559 내용이 없다고? !!
{}



 24%|███████████████████                                                            | 478/1980 [16:27<47:11,  1.89s/it]

588 내용이 없다고? !!
{}



 24%|███████████████████▎                                                           | 483/1980 [16:37<50:50,  2.04s/it]

595 내용이 없다고? !!
{}



 25%|███████████████████▋                                                           | 492/1980 [16:56<48:30,  1.96s/it]

606 내용이 없다고? !!
{}



 25%|███████████████████▊                                                           | 498/1980 [17:11<58:00,  2.35s/it]

613 내용이 없다고? !!
{}



 25%|███████████████████▉                                                           | 501/1980 [17:17<48:26,  1.97s/it]

616 내용이 없다고? !!
{}



 25%|████████████████████                                                           | 502/1980 [17:18<46:59,  1.91s/it]

618 내용이 없다고? !!
{}



 26%|████████████████████▏                                                          | 506/1980 [17:27<50:38,  2.06s/it]

622 내용이 없다고? !!
{}



 26%|████████████████████▏                                                          | 507/1980 [17:30<55:31,  2.26s/it]

623 내용이 없다고? !!
{}



 27%|█████████████████████▏                                                         | 532/1980 [18:21<48:37,  2.01s/it]

661 내용이 없다고? !!
{}



 28%|█████████████████████▋                                                         | 545/1980 [18:46<44:57,  1.88s/it]

685 내용이 없다고? !!
{}



 28%|█████████████████████▊                                                         | 547/1980 [18:49<36:45,  1.54s/it]

690 내용이 없다고? !!
{}



 28%|█████████████████████▉                                                         | 551/1980 [18:58<49:48,  2.09s/it]

700 내용이 없다고? !!
{}



 28%|██████████████████████▎                                                        | 559/1980 [19:14<47:44,  2.02s/it]

716 내용이 없다고? !!
{}



 28%|██████████████████████▍                                                        | 561/1980 [19:18<44:01,  1.86s/it]

721 내용이 없다고? !!
{}



 28%|██████████████████████▌                                                        | 564/1980 [19:23<43:31,  1.84s/it]

729 내용이 없다고? !!
{}



 29%|██████████████████████▉                                                        | 574/1980 [19:43<44:54,  1.92s/it]

740 내용이 없다고? !!
{}



 29%|███████████████████████                                                        | 578/1980 [19:54<59:04,  2.53s/it]

747 내용이 없다고? !!
{}



 29%|███████████████████████▎                                                       | 583/1980 [20:05<54:44,  2.35s/it]

754 내용이 없다고? !!
{}



 30%|███████████████████████▎                                                       | 585/1980 [20:09<52:14,  2.25s/it]

758 내용이 없다고? !!
{}



 30%|███████████████████████▍                                                       | 587/1980 [20:13<49:46,  2.14s/it]

761 내용이 없다고? !!
{}



 30%|███████████████████████▋                                                       | 593/1980 [20:24<43:38,  1.89s/it]

768 내용이 없다고? !!
{}



 30%|███████████████████████▊                                                       | 596/1980 [20:31<48:00,  2.08s/it]

771 내용이 없다고? !!
{}



 31%|████████████████████████▏                                                      | 607/1980 [20:50<41:49,  1.83s/it]

786 내용이 없다고? !!
{}



 31%|████████████████████████▍                                                      | 612/1980 [20:59<40:43,  1.79s/it]

791 내용이 없다고? !!
{}



 31%|████████████████████████▌                                                      | 615/1980 [21:06<46:25,  2.04s/it]

795 내용이 없다고? !!
{}



 31%|████████████████████████▌                                                      | 616/1980 [21:07<43:35,  1.92s/it]

797 내용이 없다고? !!
{}



 31%|████████████████████████▊                                                      | 621/1980 [21:17<42:57,  1.90s/it]

802 내용이 없다고? !!
{}



 31%|████████████████████████▊                                                      | 622/1980 [21:19<43:35,  1.93s/it]

803 내용이 없다고? !!
{}



 32%|█████████████████████████                                                      | 629/1980 [21:30<36:33,  1.62s/it]

811 내용이 없다고? !!
{}



 32%|█████████████████████████▏                                                     | 630/1980 [21:31<36:28,  1.62s/it]

812 내용이 없다고? !!
{}



 32%|█████████████████████████▍                                                     | 636/1980 [21:45<51:34,  2.30s/it]

818 내용이 없다고? !!
{}



 32%|█████████████████████████▍                                                     | 638/1980 [21:50<53:31,  2.39s/it]

821 내용이 없다고? !!
{}



 32%|█████████████████████████▌                                                     | 640/1980 [21:54<49:57,  2.24s/it]

824 내용이 없다고? !!
{}



 32%|█████████████████████████▌                                                     | 642/1980 [21:57<43:17,  1.94s/it]

828 내용이 없다고? !!
{}



 33%|█████████████████████████▋                                                     | 645/1980 [22:04<44:29,  2.00s/it]

831 내용이 없다고? !!
{}



 33%|█████████████████████████▊                                                     | 647/1980 [22:08<44:45,  2.01s/it]

833 내용이 없다고? !!
{}



 33%|█████████████████████████▊                                                     | 648/1980 [22:09<41:41,  1.88s/it]

834 내용이 없다고? !!
{}



 33%|█████████████████████████▉                                                     | 651/1980 [22:17<49:45,  2.25s/it]

837 내용이 없다고? !!
{}



 33%|██████████████████████████                                                     | 652/1980 [22:19<47:38,  2.15s/it]

838 내용이 없다고? !!
{}



 33%|██████████████████████████                                                     | 653/1980 [22:21<46:22,  2.10s/it]

839 내용이 없다고? !!
{}



 33%|██████████████████████████▏                                                    | 655/1980 [22:25<46:37,  2.11s/it]

841 내용이 없다고? !!
{}



 33%|██████████████████████████▎                                                    | 658/1980 [22:31<45:48,  2.08s/it]

846 내용이 없다고? !!
{}



 33%|██████████████████████████▍                                                    | 662/1980 [22:39<42:08,  1.92s/it]

850 내용이 없다고? !!
{}



 33%|██████████████████████████▍                                                    | 663/1980 [22:41<40:55,  1.86s/it]

851 내용이 없다고? !!
{}



 34%|██████████████████████████▋                                                    | 669/1980 [22:50<35:22,  1.62s/it]

857 내용이 없다고? !!
{}



 34%|██████████████████████████▊                                                    | 673/1980 [22:59<44:48,  2.06s/it]

863 내용이 없다고? !!
{}



 34%|██████████████████████████▉                                                    | 675/1980 [23:04<45:53,  2.11s/it]

865 내용이 없다고? !!
{}



 34%|███████████████████████████▎                                                   | 683/1980 [23:17<38:55,  1.80s/it]

873 내용이 없다고? !!
{}



 35%|███████████████████████████▌                                                   | 692/1980 [23:36<42:23,  1.97s/it]

883 내용이 없다고? !!
{}



 35%|███████████████████████████▊                                                   | 698/1980 [23:48<40:47,  1.91s/it]

890 내용이 없다고? !!
{}



 36%|████████████████████████████                                                   | 704/1980 [23:59<38:38,  1.82s/it]

896 내용이 없다고? !!
{}



 36%|████████████████████████████▏                                                  | 708/1980 [24:06<37:45,  1.78s/it]

901 내용이 없다고? !!
{}



 36%|████████████████████████████▎                                                  | 710/1980 [24:10<38:21,  1.81s/it]

903 내용이 없다고? !!
{}



 36%|████████████████████████████▌                                                  | 716/1980 [24:20<36:53,  1.75s/it]

911 내용이 없다고? !!
{}



 36%|████████████████████████████▊                                                  | 721/1980 [24:30<40:08,  1.91s/it]

917 내용이 없다고? !!
{}



 36%|████████████████████████████▊                                                  | 722/1980 [24:32<41:42,  1.99s/it]

919 내용이 없다고? !!
{}



 37%|█████████████████████████████                                                  | 729/1980 [24:48<46:24,  2.23s/it]

927 내용이 없다고? !!
{}



 37%|█████████████████████████████▍                                                 | 739/1980 [25:08<43:09,  2.09s/it]

948 내용이 없다고? !!
{}



 37%|█████████████████████████████▌                                                 | 740/1980 [25:10<40:14,  1.95s/it]

949 내용이 없다고? !!
{}



 37%|█████████████████████████████▌                                                 | 742/1980 [25:14<40:03,  1.94s/it]

952 내용이 없다고? !!
{}



 38%|█████████████████████████████▋                                                 | 744/1980 [25:18<42:39,  2.07s/it]

954 내용이 없다고? !!
{}



 38%|█████████████████████████████▋                                                 | 745/1980 [25:20<43:34,  2.12s/it]

955 내용이 없다고? !!
{}



 38%|█████████████████████████████▊                                                 | 746/1980 [25:23<44:24,  2.16s/it]

956 내용이 없다고? !!
{}



 38%|█████████████████████████████▉                                                 | 751/1980 [25:35<49:26,  2.41s/it]

962 내용이 없다고? !!
{}



 38%|██████████████████████████████                                                 | 752/1980 [25:37<46:05,  2.25s/it]

964 내용이 없다고? !!
{}



 38%|██████████████████████████████                                                 | 753/1980 [25:38<43:15,  2.12s/it]

965 내용이 없다고? !!
{}



 38%|██████████████████████████████▏                                                | 756/1980 [25:45<48:09,  2.36s/it]

968 내용이 없다고? !!
{}



 39%|██████████████████████████████▍                                                | 763/1980 [26:01<43:51,  2.16s/it]

975 내용이 없다고? !!
{}



 39%|██████████████████████████████▊                                                | 771/1980 [26:18<41:49,  2.08s/it]

983 내용이 없다고? !!
{}



 40%|███████████████████████████████▏                                               | 783/1980 [26:43<40:16,  2.02s/it]

996 내용이 없다고? !!
{}



 40%|███████████████████████████████▍                                               | 787/1980 [26:51<40:15,  2.02s/it]

1000 내용이 없다고? !!
{}



 40%|███████████████████████████████▍                                               | 789/1980 [26:55<37:25,  1.89s/it]

1002 내용이 없다고? !!
{}



 40%|███████████████████████████████▌                                               | 790/1980 [26:57<36:15,  1.83s/it]

1003 내용이 없다고? !!
{}



 40%|███████████████████████████████▊                                               | 796/1980 [27:08<37:38,  1.91s/it]

1009 내용이 없다고? !!
{}



 41%|████████████████████████████████▎                                              | 810/1980 [27:34<35:51,  1.84s/it]

1026 내용이 없다고? !!
{}



 42%|████████████████████████████████▉                                              | 826/1980 [28:07<39:31,  2.05s/it]

1045 내용이 없다고? !!
{}



 42%|████████████████████████████████▉                                              | 827/1980 [28:08<36:29,  1.90s/it]

1046 내용이 없다고? !!
{}



 42%|█████████████████████████████████▏                                             | 833/1980 [28:19<36:54,  1.93s/it]

1052 내용이 없다고? !!
{}



 42%|█████████████████████████████████▌                                             | 840/1980 [28:35<38:06,  2.01s/it]

1059 내용이 없다고? !!
{}



 43%|████████████████████████████████▉                                            | 847/1980 [28:59<1:16:54,  4.07s/it]

1069 내용이 없다고? !!
{}



 43%|██████████████████████████████████                                             | 853/1980 [29:12<46:28,  2.47s/it]

1076 내용이 없다고? !!
{}



 43%|██████████████████████████████████                                             | 855/1980 [29:15<38:00,  2.03s/it]

1078 내용이 없다고? !!
{}



 43%|██████████████████████████████████▎                                            | 860/1980 [29:28<42:24,  2.27s/it]

1083 내용이 없다고? !!
{}



 44%|██████████████████████████████████▍                                            | 863/1980 [29:34<38:51,  2.09s/it]

1087 내용이 없다고? !!
{}



 44%|██████████████████████████████████▌                                            | 867/1980 [29:41<32:49,  1.77s/it]

1091 내용이 없다고? !!
{}



 44%|██████████████████████████████████▊                                            | 874/1980 [29:55<36:51,  2.00s/it]

1098 내용이 없다고? !!
{}



 44%|██████████████████████████████████▉                                            | 876/1980 [29:59<36:44,  2.00s/it]

1100 내용이 없다고? !!
{}



 44%|███████████████████████████████████                                            | 880/1980 [30:07<36:57,  2.02s/it]

1104 내용이 없다고? !!
{}



 45%|███████████████████████████████████▏                                           | 882/1980 [30:11<37:33,  2.05s/it]

1106 내용이 없다고? !!
{}



 45%|███████████████████████████████████▎                                           | 884/1980 [30:15<34:59,  1.92s/it]

1108 내용이 없다고? !!
{}



 45%|███████████████████████████████████▍                                           | 889/1980 [30:24<33:00,  1.82s/it]

1113 내용이 없다고? !!
{}



 45%|███████████████████████████████████▋                                           | 894/1980 [30:34<35:53,  1.98s/it]

1118 내용이 없다고? !!
{}



 45%|███████████████████████████████████▊                                           | 897/1980 [30:39<32:44,  1.81s/it]

1121 내용이 없다고? !!
{}



 45%|███████████████████████████████████▉                                           | 900/1980 [30:45<34:45,  1.93s/it]

1124 내용이 없다고? !!
{}



 46%|███████████████████████████████████▉                                           | 901/1980 [30:48<36:14,  2.02s/it]

1125 내용이 없다고? !!
{}



 46%|████████████████████████████████████▏                                          | 908/1980 [31:03<36:14,  2.03s/it]

1132 내용이 없다고? !!
{}



 46%|████████████████████████████████████▎                                          | 909/1980 [31:05<33:47,  1.89s/it]

1133 내용이 없다고? !!
{}



 47%|████████████████████████████████████▊                                          | 923/1980 [31:32<39:25,  2.24s/it]

1149 내용이 없다고? !!
{}



 47%|█████████████████████████████████████▏                                         | 932/1980 [31:50<33:48,  1.94s/it]

1158 내용이 없다고? !!
{}



 48%|█████████████████████████████████████▌                                         | 942/1980 [32:10<33:33,  1.94s/it]

1168 내용이 없다고? !!
{}



 48%|██████████████████████████████████████▏                                        | 956/1980 [32:38<37:55,  2.22s/it]

1183 내용이 없다고? !!
{}



 48%|██████████████████████████████████████▏                                        | 958/1980 [32:43<38:15,  2.25s/it]

1185 내용이 없다고? !!
{}



 49%|██████████████████████████████████████▍                                        | 963/1980 [32:53<33:04,  1.95s/it]

1190 내용이 없다고? !!
{}



 49%|██████████████████████████████████████▌                                        | 965/1980 [32:57<32:53,  1.94s/it]

1192 내용이 없다고? !!
{}



 49%|██████████████████████████████████████▉                                        | 977/1980 [33:22<38:16,  2.29s/it]

1205 내용이 없다고? !!
{}



 49%|███████████████████████████████████████                                        | 980/1980 [33:27<34:22,  2.06s/it]

1208 내용이 없다고? !!
{}



 50%|███████████████████████████████████████▏                                       | 981/1980 [33:29<32:25,  1.95s/it]

1209 내용이 없다고? !!
{}



 50%|███████████████████████████████████████▏                                       | 982/1980 [33:31<31:34,  1.90s/it]

1210 내용이 없다고? !!
{}



 50%|███████████████████████████████████████▏                                       | 983/1980 [33:33<30:32,  1.84s/it]

1211 내용이 없다고? !!
{}



 50%|███████████████████████████████████████▍                                       | 988/1980 [33:42<31:42,  1.92s/it]

1216 내용이 없다고? !!
{}



 50%|███████████████████████████████████████▌                                       | 990/1980 [33:48<39:35,  2.40s/it]

1218 내용이 없다고? !!
{}



 51%|███████████████████████████████████████▋                                      | 1007/1980 [34:25<35:39,  2.20s/it]

1237 내용이 없다고? !!
{}



 51%|███████████████████████████████████████▋                                      | 1008/1980 [34:27<32:42,  2.02s/it]

1238 내용이 없다고? !!
{}



 51%|███████████████████████████████████████▉                                      | 1013/1980 [34:38<35:37,  2.21s/it]

1245 내용이 없다고? !!
{}



 52%|████████████████████████████████████████▏                                     | 1020/1980 [34:51<28:29,  1.78s/it]

1253 내용이 없다고? !!
{}



 52%|████████████████████████████████████████▎                                     | 1022/1980 [34:55<29:52,  1.87s/it]

1255 내용이 없다고? !!
{}



 52%|████████████████████████████████████████▍                                     | 1025/1980 [35:00<28:25,  1.79s/it]

1258 내용이 없다고? !!
{}



 52%|████████████████████████████████████████▍                                     | 1027/1980 [35:05<34:37,  2.18s/it]

1260 내용이 없다고? !!
{}



 52%|████████████████████████████████████████▊                                     | 1035/1980 [35:21<33:02,  2.10s/it]

1268 내용이 없다고? !!
{}



 52%|████████████████████████████████████████▊                                     | 1036/1980 [35:24<33:29,  2.13s/it]

1269 내용이 없다고? !!
{}



 53%|█████████████████████████████████████████                                     | 1043/1980 [35:40<36:59,  2.37s/it]

1277 내용이 없다고? !!
{}



 53%|█████████████████████████████████████████▏                                    | 1044/1980 [35:42<33:52,  2.17s/it]

1278 내용이 없다고? !!
{}



 53%|█████████████████████████████████████████▍                                    | 1051/1980 [35:57<35:44,  2.31s/it]

1285 내용이 없다고? !!
{}



 53%|█████████████████████████████████████████▌                                    | 1056/1980 [36:09<36:13,  2.35s/it]

1290 내용이 없다고? !!
{}



 54%|█████████████████████████████████████████▉                                    | 1065/1980 [36:28<35:19,  2.32s/it]

1300 내용이 없다고? !!
{}



 54%|██████████████████████████████████████████▎                                   | 1075/1980 [36:50<29:21,  1.95s/it]

1310 내용이 없다고? !!
{}



 55%|██████████████████████████████████████████▋                                   | 1084/1980 [37:07<28:09,  1.89s/it]

1321 내용이 없다고? !!
{}



 55%|██████████████████████████████████████████▋                                   | 1085/1980 [37:09<27:32,  1.85s/it]

1322 내용이 없다고? !!
{}



 55%|██████████████████████████████████████████▉                                   | 1089/1980 [37:18<31:09,  2.10s/it]

1327 내용이 없다고? !!
{}



 55%|██████████████████████████████████████████▉                                   | 1090/1980 [37:20<29:29,  1.99s/it]

1328 내용이 없다고? !!
{}



 56%|███████████████████████████████████████████▍                                  | 1103/1980 [37:48<29:38,  2.03s/it]

1341 내용이 없다고? !!
{}



 56%|███████████████████████████████████████████▌                                  | 1107/1980 [37:57<30:59,  2.13s/it]

1345 내용이 없다고? !!
{}



 56%|███████████████████████████████████████████▋                                  | 1109/1980 [38:01<29:32,  2.03s/it]

1347 내용이 없다고? !!
{}



 56%|███████████████████████████████████████████▊                                  | 1113/1980 [38:07<25:58,  1.80s/it]

1351 내용이 없다고? !!
{}



 56%|███████████████████████████████████████████▉                                  | 1114/1980 [38:09<25:57,  1.80s/it]

1352 내용이 없다고? !!
{}



 57%|████████████████████████████████████████████                                  | 1119/1980 [38:20<28:00,  1.95s/it]

1357 내용이 없다고? !!
{}



 57%|████████████████████████████████████████████                                  | 1120/1980 [38:22<30:59,  2.16s/it]

1358 내용이 없다고? !!
{}



 57%|████████████████████████████████████████████▏                                 | 1121/1980 [38:25<32:18,  2.26s/it]

1359 내용이 없다고? !!
{}



 57%|████████████████████████████████████████████▏                                 | 1123/1980 [38:29<31:03,  2.17s/it]

1361 내용이 없다고? !!
{}



 57%|████████████████████████████████████████████▎                                 | 1126/1980 [38:35<29:52,  2.10s/it]

1365 내용이 없다고? !!
{}



 58%|████████████████████████████████████████████▉                                 | 1141/1980 [39:05<27:58,  2.00s/it]

1380 내용이 없다고? !!
{}



 58%|█████████████████████████████████████████████                                 | 1144/1980 [39:12<28:50,  2.07s/it]

1383 내용이 없다고? !!
{}



 58%|█████████████████████████████████████████████▏                                | 1146/1980 [39:15<26:17,  1.89s/it]

1385 내용이 없다고? !!
{}



 58%|█████████████████████████████████████████████▎                                | 1149/1980 [39:21<27:18,  1.97s/it]

1390 내용이 없다고? !!
{}



 58%|█████████████████████████████████████████████▌                                | 1158/1980 [39:40<32:03,  2.34s/it]

1400 내용이 없다고? !!
{}



 59%|█████████████████████████████████████████████▋                                | 1160/1980 [39:48<40:52,  2.99s/it]

1402 내용이 없다고? !!
{}



 59%|█████████████████████████████████████████████▊                                | 1162/1980 [39:53<34:49,  2.55s/it]

1404 내용이 없다고? !!
{}



 59%|█████████████████████████████████████████████▊                                | 1163/1980 [39:54<31:11,  2.29s/it]

1405 내용이 없다고? !!
{}



 59%|██████████████████████████████████████████████▏                               | 1173/1980 [40:13<22:25,  1.67s/it]

1415 내용이 없다고? !!
{}



 59%|██████████████████████████████████████████████▎                               | 1177/1980 [40:22<27:30,  2.06s/it]

1419 내용이 없다고? !!
{}



 59%|██████████████████████████████████████████████▍                               | 1178/1980 [40:24<27:03,  2.02s/it]

1420 내용이 없다고? !!
{}



 60%|██████████████████████████████████████████████▌                               | 1183/1980 [40:34<29:27,  2.22s/it]

1426 내용이 없다고? !!
{}



 60%|██████████████████████████████████████████████▋                               | 1185/1980 [40:39<30:22,  2.29s/it]

1428 내용이 없다고? !!
{}



 60%|██████████████████████████████████████████████▊                               | 1188/1980 [40:45<27:22,  2.07s/it]

1431 내용이 없다고? !!
{}



 60%|███████████████████████████████████████████████                               | 1195/1980 [40:59<28:23,  2.17s/it]

1438 내용이 없다고? !!
{}



 60%|███████████████████████████████████████████████                               | 1196/1980 [41:02<28:34,  2.19s/it]

1439 내용이 없다고? !!
{}



 61%|███████████████████████████████████████████████▎                              | 1202/1980 [41:14<25:52,  2.00s/it]

1445 내용이 없다고? !!
{}



 61%|███████████████████████████████████████████████▍                              | 1203/1980 [41:16<27:02,  2.09s/it]

1446 내용이 없다고? !!
{}



 61%|███████████████████████████████████████████████▋                              | 1210/1980 [41:30<27:03,  2.11s/it]

1454 내용이 없다고? !!
{}



 62%|████████████████████████████████████████████████                              | 1221/1980 [41:53<25:12,  1.99s/it]

1465 내용이 없다고? !!
{}



 62%|████████████████████████████████████████████████▏                             | 1224/1980 [41:59<25:06,  1.99s/it]

1468 내용이 없다고? !!
{}



 62%|████████████████████████████████████████████████▍                             | 1229/1980 [42:11<30:03,  2.40s/it]

1481 내용이 없다고? !!
{}



 62%|████████████████████████████████████████████████▌                             | 1234/1980 [42:20<23:43,  1.91s/it]

1510 내용이 없다고? !!
{}



 63%|████████████████████████████████████████████████▊                             | 1240/1980 [42:33<26:40,  2.16s/it]

1517 내용이 없다고? !!
{}



 63%|████████████████████████████████████████████████▉                             | 1242/1980 [42:42<37:18,  3.03s/it]

1520 내용이 없다고? !!
{}



 64%|█████████████████████████████████████████████████▋                            | 1262/1980 [43:23<27:29,  2.30s/it]

1561 내용이 없다고? !!
{}



 64%|█████████████████████████████████████████████████▉                            | 1269/1980 [43:35<18:02,  1.52s/it]

1571 내용이 없다고? !!
{}



 64%|██████████████████████████████████████████████████                            | 1271/1980 [43:38<19:03,  1.61s/it]

1575 내용이 없다고? !!
{}



 64%|██████████████████████████████████████████████████▎                           | 1276/1980 [43:47<21:09,  1.80s/it]

1582 내용이 없다고? !!
{}



 65%|██████████████████████████████████████████████████▋                           | 1288/1980 [44:09<21:17,  1.85s/it]

1598 내용이 없다고? !!
{}



 65%|██████████████████████████████████████████████████▊                           | 1290/1980 [44:12<20:04,  1.75s/it]

1600 내용이 없다고? !!
{}



 66%|███████████████████████████████████████████████████▋                          | 1312/1980 [44:53<19:53,  1.79s/it]

1633 내용이 없다고? !!
{}



 67%|███████████████████████████████████████████████████▉                          | 1318/1980 [45:04<18:48,  1.71s/it]

1639 내용이 없다고? !!
{}



 67%|████████████████████████████████████████████████████▎                         | 1328/1980 [45:27<27:09,  2.50s/it]

1653 내용이 없다고? !!
{}



 67%|████████████████████████████████████████████████████▌                         | 1335/1980 [45:39<19:56,  1.86s/it]

1663 내용이 없다고? !!
{}



 68%|████████████████████████████████████████████████████▋                         | 1339/1980 [45:48<23:40,  2.22s/it]

1668 내용이 없다고? !!
{}



 68%|█████████████████████████████████████████████████████                         | 1346/1980 [46:03<21:28,  2.03s/it]

1676 내용이 없다고? !!
{}



 68%|█████████████████████████████████████████████████████▏                        | 1349/1980 [46:09<21:00,  2.00s/it]

1679 내용이 없다고? !!
{}



 69%|██████████████████████████████████████████████████████                        | 1371/1980 [46:53<19:45,  1.95s/it]

1706 내용이 없다고? !!
{}



 70%|██████████████████████████████████████████████████████▋                       | 1389/1980 [47:31<19:21,  1.97s/it]

1724 내용이 없다고? !!
{}



 71%|███████████████████████████████████████████████████████▏                      | 1401/1980 [47:55<19:48,  2.05s/it]

1736 내용이 없다고? !!
{}



 71%|███████████████████████████████████████████████████████▍                      | 1407/1980 [48:08<19:54,  2.09s/it]

1748 내용이 없다고? !!
{}



 71%|███████████████████████████████████████████████████████▍                      | 1408/1980 [48:10<20:02,  2.10s/it]

1749 내용이 없다고? !!
{}



 72%|███████████████████████████████████████████████████████▉                      | 1421/1980 [48:40<21:02,  2.26s/it]

1782 내용이 없다고? !!
{}



 73%|████████████████████████████████████████████████████████▋                     | 1438/1980 [49:17<18:24,  2.04s/it]

1816 내용이 없다고? !!
{}



 73%|████████████████████████████████████████████████████████▉                     | 1444/1980 [49:28<15:50,  1.77s/it]

1826 내용이 없다고? !!
{}



 73%|█████████████████████████████████████████████████████████                     | 1447/1980 [49:34<17:19,  1.95s/it]

1830 내용이 없다고? !!
{}



 74%|█████████████████████████████████████████████████████████▎                    | 1456/1980 [49:53<21:28,  2.46s/it]

1848 내용이 없다고? !!
{}



 74%|█████████████████████████████████████████████████████████▌                    | 1460/1980 [50:03<20:52,  2.41s/it]

1857 내용이 없다고? !!
{}



 74%|█████████████████████████████████████████████████████████▌                    | 1461/1980 [50:05<20:10,  2.33s/it]

1860 내용이 없다고? !!
{}



 74%|█████████████████████████████████████████████████████████▌                    | 1462/1980 [50:07<18:24,  2.13s/it]

1861 내용이 없다고? !!
{}



 74%|█████████████████████████████████████████████████████████▋                    | 1465/1980 [50:12<16:40,  1.94s/it]

1864 내용이 없다고? !!
{}



 74%|██████████████████████████████████████████████████████████                    | 1475/1980 [50:33<19:34,  2.33s/it]

1879 내용이 없다고? !!
{}



 75%|██████████████████████████████████████████████████████████▏                   | 1476/1980 [50:35<17:36,  2.10s/it]

1880 내용이 없다고? !!
{}



 75%|██████████████████████████████████████████████████████████▏                   | 1478/1980 [50:38<16:04,  1.92s/it]

1882 내용이 없다고? !!
{}



 75%|██████████████████████████████████████████████████████████▌                   | 1485/1980 [50:55<16:40,  2.02s/it]

1892 내용이 없다고? !!
{}



 75%|██████████████████████████████████████████████████████████▋                   | 1490/1980 [51:07<17:01,  2.08s/it]

1901 내용이 없다고? !!
{}



 76%|███████████████████████████████████████████████████████████                   | 1498/1980 [51:26<20:05,  2.50s/it]

1912 내용이 없다고? !!
{}



 76%|███████████████████████████████████████████████████████████▏                  | 1502/1980 [51:34<16:15,  2.04s/it]

1917 내용이 없다고? !!
{}



 76%|███████████████████████████████████████████████████████████▍                  | 1510/1980 [51:52<17:49,  2.28s/it]

1925 내용이 없다고? !!
{}



 76%|███████████████████████████████████████████████████████████▌                  | 1512/1980 [51:55<15:26,  1.98s/it]

1927 내용이 없다고? !!
{}



 77%|████████████████████████████████████████████████████████████                  | 1525/1980 [52:21<15:42,  2.07s/it]

1941 내용이 없다고? !!
{}



 77%|████████████████████████████████████████████████████████████▏                 | 1529/1980 [52:29<16:00,  2.13s/it]

1946 내용이 없다고? !!
{}



 77%|████████████████████████████████████████████████████████████▎                 | 1530/1980 [52:32<17:07,  2.28s/it]

1947 내용이 없다고? !!
{}



 77%|████████████████████████████████████████████████████████████▎                 | 1531/1980 [52:34<17:08,  2.29s/it]

1948 내용이 없다고? !!
{}



 77%|████████████████████████████████████████████████████████████▎                 | 1532/1980 [52:36<15:48,  2.12s/it]

1949 내용이 없다고? !!
{}



 77%|████████████████████████████████████████████████████████████▍                 | 1534/1980 [52:41<16:23,  2.21s/it]

1951 내용이 없다고? !!
{}



 78%|████████████████████████████████████████████████████████████▍                 | 1535/1980 [52:42<15:20,  2.07s/it]

1953 내용이 없다고? !!
{}



 78%|████████████████████████████████████████████████████████████▌                 | 1538/1980 [52:49<15:28,  2.10s/it]

1957 내용이 없다고? !!
{}



 78%|████████████████████████████████████████████████████████████▋                 | 1542/1980 [52:58<15:57,  2.19s/it]

1962 내용이 없다고? !!
{}



 78%|████████████████████████████████████████████████████████████▊                 | 1543/1980 [53:00<16:12,  2.23s/it]

1963 내용이 없다고? !!
{}



 78%|████████████████████████████████████████████████████████████▉                 | 1548/1980 [53:12<17:52,  2.48s/it]

1972 내용이 없다고? !!
{}



 78%|█████████████████████████████████████████████████████████████                 | 1550/1980 [53:17<16:50,  2.35s/it]

1974 내용이 없다고? !!
{}



 78%|█████████████████████████████████████████████████████████████                 | 1551/1980 [53:18<15:31,  2.17s/it]

1975 내용이 없다고? !!
{}



 78%|█████████████████████████████████████████████████████████████▏                | 1554/1980 [53:24<14:06,  1.99s/it]

1978 내용이 없다고? !!
{}



 79%|█████████████████████████████████████████████████████████████▍                | 1559/1980 [53:34<13:25,  1.91s/it]

1983 내용이 없다고? !!
{}



 79%|█████████████████████████████████████████████████████████████▋                | 1565/1980 [53:48<15:58,  2.31s/it]

1990 내용이 없다고? !!
{}



 79%|█████████████████████████████████████████████████████████████▉                | 1573/1980 [54:04<12:30,  1.84s/it]

2001 내용이 없다고? !!
{}



 80%|██████████████████████████████████████████████████████████████▏               | 1578/1980 [54:16<14:10,  2.12s/it]

2006 내용이 없다고? !!
{}



 80%|██████████████████████████████████████████████████████████████▎               | 1582/1980 [54:23<12:14,  1.85s/it]

2011 내용이 없다고? !!
{}



 80%|██████████████████████████████████████████████████████████████▋               | 1590/1980 [54:37<11:42,  1.80s/it]

2019 내용이 없다고? !!
{}



 81%|██████████████████████████████████████████████████████████████▊               | 1596/1980 [54:51<13:50,  2.16s/it]

2025 내용이 없다고? !!
{}



 81%|██████████████████████████████████████████████████████████████▉               | 1598/1980 [54:55<13:35,  2.13s/it]

2028 내용이 없다고? !!
{}



 81%|██████████████████████████████████████████████████████████████▉               | 1599/1980 [54:56<12:28,  1.96s/it]

2029 내용이 없다고? !!
{}



 81%|███████████████████████████████████████████████████████████████               | 1602/1980 [55:02<12:04,  1.92s/it]

2032 내용이 없다고? !!
{}



 81%|███████████████████████████████████████████████████████████████▍              | 1611/1980 [55:19<10:59,  1.79s/it]

2042 내용이 없다고? !!
{}



 81%|███████████████████████████████████████████████████████████████▌              | 1612/1980 [55:21<11:17,  1.84s/it]

2044 내용이 없다고? !!
{}



 82%|███████████████████████████████████████████████████████████████▋              | 1618/1980 [55:35<13:32,  2.24s/it]

2050 내용이 없다고? !!
{}



 82%|███████████████████████████████████████████████████████████████▊              | 1620/1980 [55:40<14:16,  2.38s/it]

2052 내용이 없다고? !!
{}



 82%|████████████████████████████████████████████████████████████████              | 1626/1980 [55:51<11:55,  2.02s/it]

2059 내용이 없다고? !!
{}



 82%|████████████████████████████████████████████████████████████████▏             | 1630/1980 [55:58<10:41,  1.83s/it]

2064 내용이 없다고? !!
{}



 82%|████████████████████████████████████████████████████████████████▎             | 1632/1980 [56:02<10:44,  1.85s/it]

2066 내용이 없다고? !!
{}



 83%|████████████████████████████████████████████████████████████████▎             | 1634/1980 [56:06<11:38,  2.02s/it]

2068 내용이 없다고? !!
{}



 83%|████████████████████████████████████████████████████████████████▌             | 1640/1980 [56:19<11:24,  2.01s/it]

2076 내용이 없다고? !!
{}



 83%|████████████████████████████████████████████████████████████████▊             | 1644/1980 [56:26<10:59,  1.96s/it]

2080 내용이 없다고? !!
{}



 84%|█████████████████████████████████████████████████████████████████▏            | 1656/1980 [56:51<10:36,  1.97s/it]

2094 내용이 없다고? !!
{}



 84%|█████████████████████████████████████████████████████████████████▍            | 1661/1980 [57:03<13:18,  2.50s/it]

2099 내용이 없다고? !!
{}



 84%|█████████████████████████████████████████████████████████████████▌            | 1663/1980 [57:06<11:18,  2.14s/it]

2101 내용이 없다고? !!
{}



 84%|█████████████████████████████████████████████████████████████████▋            | 1668/1980 [57:16<10:10,  1.96s/it]

2107 내용이 없다고? !!
{}



 85%|██████████████████████████████████████████████████████████████████            | 1677/1980 [57:36<10:34,  2.09s/it]

2117 내용이 없다고? !!
{}



 85%|██████████████████████████████████████████████████████████████████▎           | 1683/1980 [57:45<08:56,  1.81s/it]

2123 내용이 없다고? !!
{}



 86%|██████████████████████████████████████████████████████████████████▊           | 1697/1980 [58:18<09:56,  2.11s/it]

2138 내용이 없다고? !!
{}



 86%|███████████████████████████████████████████████████████████████████           | 1703/1980 [58:32<11:03,  2.40s/it]

2144 내용이 없다고? !!
{}



 86%|███████████████████████████████████████████████████████████████████▏          | 1704/1980 [58:34<10:06,  2.20s/it]

2145 내용이 없다고? !!
{}



 86%|███████████████████████████████████████████████████████████████████▏          | 1706/1980 [58:38<09:49,  2.15s/it]

2147 내용이 없다고? !!
{}



 86%|███████████████████████████████████████████████████████████████████▍          | 1711/1980 [58:47<08:19,  1.86s/it]

2152 내용이 없다고? !!
{}



 87%|███████████████████████████████████████████████████████████████████▉          | 1723/1980 [59:14<10:04,  2.35s/it]

2165 내용이 없다고? !!
{}



 87%|███████████████████████████████████████████████████████████████████▉          | 1724/1980 [59:16<09:35,  2.25s/it]

2166 내용이 없다고? !!
{}



 88%|████████████████████████████████████████████████████████████████████▍         | 1736/1980 [59:43<08:38,  2.12s/it]

2178 내용이 없다고? !!
{}



 88%|████████████████████████████████████████████████████████████████████▌         | 1742/1980 [59:57<09:29,  2.39s/it]

2185 내용이 없다고? !!
{}



 88%|███████████████████████████████████████████████████████████████████▏        | 1750/1980 [1:00:12<08:18,  2.17s/it]

2193 내용이 없다고? !!
{}



 88%|███████████████████████████████████████████████████████████████████▏        | 1751/1980 [1:00:14<07:34,  1.99s/it]

2194 내용이 없다고? !!
{}



 89%|███████████████████████████████████████████████████████████████████▍        | 1758/1980 [1:00:29<07:07,  1.93s/it]

2201 내용이 없다고? !!
{}



 89%|███████████████████████████████████████████████████████████████████▌        | 1761/1980 [1:00:34<06:18,  1.73s/it]

2204 내용이 없다고? !!
{}



 89%|███████████████████████████████████████████████████████████████████▋        | 1763/1980 [1:00:38<06:54,  1.91s/it]

2206 내용이 없다고? !!
{}



 89%|███████████████████████████████████████████████████████████████████▊        | 1766/1980 [1:00:44<07:04,  1.98s/it]

2209 내용이 없다고? !!
{}



 89%|███████████████████████████████████████████████████████████████████▊        | 1767/1980 [1:00:46<07:12,  2.03s/it]

2210 내용이 없다고? !!
{}



 89%|███████████████████████████████████████████████████████████████████▉        | 1770/1980 [1:00:54<08:24,  2.40s/it]

2213 내용이 없다고? !!
{}



 89%|████████████████████████████████████████████████████████████████████        | 1772/1980 [1:00:58<07:15,  2.09s/it]

2215 내용이 없다고? !!
{}



 90%|████████████████████████████████████████████████████████████████████        | 1773/1980 [1:01:00<07:12,  2.09s/it]

2216 내용이 없다고? !!
{}



 90%|████████████████████████████████████████████████████████████████████▎       | 1780/1980 [1:01:13<06:36,  1.98s/it]

2223 내용이 없다고? !!
{}



 90%|████████████████████████████████████████████████████████████████████▍       | 1784/1980 [1:01:21<06:19,  1.94s/it]

2227 내용이 없다고? !!
{}



 91%|████████████████████████████████████████████████████████████████████▉       | 1795/1980 [1:01:44<06:25,  2.08s/it]

2239 내용이 없다고? !!
{}



 91%|█████████████████████████████████████████████████████████████████████       | 1800/1980 [1:01:57<08:18,  2.77s/it]

2245 내용이 없다고? !!
{}



 91%|█████████████████████████████████████████████████████████████████████▏      | 1803/1980 [1:02:04<06:45,  2.29s/it]

2248 내용이 없다고? !!
{}



 91%|█████████████████████████████████████████████████████████████████████▎      | 1805/1980 [1:02:07<06:02,  2.07s/it]

2251 내용이 없다고? !!
{}



 91%|█████████████████████████████████████████████████████████████████████▍      | 1808/1980 [1:02:13<05:45,  2.01s/it]

2255 내용이 없다고? !!
{}



 91%|█████████████████████████████████████████████████████████████████████▍      | 1809/1980 [1:02:15<05:32,  1.95s/it]

2256 내용이 없다고? !!
{}



 92%|█████████████████████████████████████████████████████████████████████▌      | 1812/1980 [1:02:21<05:46,  2.06s/it]

2259 내용이 없다고? !!
{}



 92%|█████████████████████████████████████████████████████████████████████▋      | 1814/1980 [1:02:25<05:37,  2.03s/it]

2261 내용이 없다고? !!
{}



 92%|█████████████████████████████████████████████████████████████████████▋      | 1816/1980 [1:02:29<05:10,  1.89s/it]

2264 내용이 없다고? !!
{}



 92%|█████████████████████████████████████████████████████████████████████▊      | 1818/1980 [1:02:32<05:06,  1.89s/it]

2266 내용이 없다고? !!
{}



 92%|██████████████████████████████████████████████████████████████████████      | 1824/1980 [1:02:47<06:16,  2.41s/it]

2274 내용이 없다고? !!
{}



 92%|██████████████████████████████████████████████████████████████████████▏     | 1827/1980 [1:02:53<05:22,  2.11s/it]

2277 내용이 없다고? !!
{}



 92%|██████████████████████████████████████████████████████████████████████▏     | 1829/1980 [1:02:57<05:22,  2.13s/it]

2279 내용이 없다고? !!
{}



 93%|██████████████████████████████████████████████████████████████████████▉     | 1847/1980 [1:03:35<04:47,  2.16s/it]

2297 내용이 없다고? !!
{}



 93%|██████████████████████████████████████████████████████████████████████▉     | 1849/1980 [1:03:39<04:52,  2.23s/it]

2304 내용이 없다고? !!
{}



 94%|███████████████████████████████████████████████████████████████████████▎    | 1858/1980 [1:04:00<04:44,  2.33s/it]

2321 내용이 없다고? !!
{}



 94%|███████████████████████████████████████████████████████████████████████▍    | 1862/1980 [1:04:09<04:08,  2.11s/it]

2328 내용이 없다고? !!
{}



 94%|███████████████████████████████████████████████████████████████████████▌    | 1863/1980 [1:04:11<04:16,  2.19s/it]

2329 내용이 없다고? !!
{}



 94%|███████████████████████████████████████████████████████████████████████▌    | 1866/1980 [1:04:17<03:49,  2.02s/it]

2333 내용이 없다고? !!
{}



 94%|███████████████████████████████████████████████████████████████████████▊    | 1871/1980 [1:04:28<03:41,  2.03s/it]

2346 내용이 없다고? !!
{}



 95%|███████████████████████████████████████████████████████████████████████▉    | 1873/1980 [1:04:32<03:39,  2.05s/it]

2350 내용이 없다고? !!
{}



 95%|████████████████████████████████████████████████████████████████████████    | 1876/1980 [1:04:37<03:12,  1.85s/it]

2355 내용이 없다고? !!
{}



 95%|████████████████████████████████████████████████████████████████████████    | 1877/1980 [1:04:39<03:19,  1.94s/it]

2356 내용이 없다고? !!
{}



 95%|████████████████████████████████████████████████████████████████████████▏   | 1880/1980 [1:04:45<03:10,  1.90s/it]

2360 내용이 없다고? !!
{}



 95%|████████████████████████████████████████████████████████████████████████▎   | 1885/1980 [1:04:57<03:40,  2.32s/it]

2365 내용이 없다고? !!
{}



 96%|████████████████████████████████████████████████████████████████████████▋   | 1895/1980 [1:05:19<03:05,  2.18s/it]

2386 내용이 없다고? !!
{}



 96%|████████████████████████████████████████████████████████████████████████▊   | 1897/1980 [1:05:23<02:47,  2.02s/it]

2388 내용이 없다고? !!
{}



 96%|█████████████████████████████████████████████████████████████████████████▏  | 1907/1980 [1:05:43<02:19,  1.92s/it]

2401 내용이 없다고? !!
{}



 96%|█████████████████████████████████████████████████████████████████████████▎  | 1910/1980 [1:05:50<02:35,  2.23s/it]

2405 내용이 없다고? !!
{}



 97%|█████████████████████████████████████████████████████████████████████████▉  | 1926/1980 [1:06:22<01:51,  2.06s/it]

2432 내용이 없다고? !!
{}



 97%|██████████████████████████████████████████████████████████████████████████  | 1930/1980 [1:06:29<01:29,  1.79s/it]

2438 내용이 없다고? !!
{}



 98%|██████████████████████████████████████████████████████████████████████████▎ | 1935/1980 [1:06:41<01:32,  2.06s/it]

2444 내용이 없다고? !!
{}



 98%|██████████████████████████████████████████████████████████████████████████▎ | 1936/1980 [1:06:43<01:29,  2.03s/it]

2445 내용이 없다고? !!
{}



 98%|██████████████████████████████████████████████████████████████████████████▋ | 1947/1980 [1:07:03<01:07,  2.06s/it]

2474 내용이 없다고? !!
{}



 99%|███████████████████████████████████████████████████████████████████████████ | 1957/1980 [1:07:25<00:52,  2.28s/it]

2494 내용이 없다고? !!
{}



 99%|███████████████████████████████████████████████████████████████████████████▍| 1964/1980 [1:07:39<00:32,  2.03s/it]

2504 내용이 없다고? !!
{}



100%|███████████████████████████████████████████████████████████████████████████▊| 1976/1980 [1:08:01<00:06,  1.56s/it]

2517 내용이 없다고? !!
{}



100%|████████████████████████████████████████████████████████████████████████████| 1980/1980 [1:08:09<00:00,  2.07s/it]

2524 내용이 없다고? !!
{}



## 3.3 json 파일 정보를 DataFrame에 매핑

In [26]:
# json 파일을 매핑할 키 정보들
df[['광고 분류2',"index"]]

광고 분류2  index
0          2      0
1          2      1
2          2      2
3          2      3
4          2      4
...      ...    ...
2520       2   2520
2521       1   2521
2522       2   2522
2523       0   2523
2524       0   2524

[2525 rows x 2 columns]

In [40]:
# jsonFilea 정보 저장
def json_labeling(x):
    index_num = int(x[0])
    label_num = int(x[1])
    load_name = f"annotation/{label_num}/{index_num}_class{label_num}.json" 
    return load_name

In [42]:
df["jsonFile"] = df[["index","광고 분류2"]].apply(lambda x: json_labeling(x), axis=1)

In [43]:
# 정보 확인
df

index                                                url  \
0         0  https://blog.naver.com/PostView.naver?blogId=e...   
1         1  https://blog.naver.com/PostView.naver?blogId=s...   
2         2  https://blog.naver.com/PostView.naver?blogId=c...   
3         3  https://blog.naver.com/PostView.naver?blogId=p...   
4         4  https://blog.naver.com/PostView.naver?blogId=a...   
...     ...                                                ...   
2520   2520  https://blog.naver.com/PostView.naver?blogId=h...   
2521   2521  https://blog.naver.com/PostView.naver?blogId=g...   
2522   2522  https://blog.naver.com/PostView.naver?blogId=b...   
2523   2523  https://blog.naver.com/PostView.naver?blogId=l...   
2524   2524  https://blog.naver.com/PostView.naver?blogId=h...   

                                               title  post_date        검색어  \
0                           무선 무릎찜질기 메이크맨 무르비2 사용 후기   20220124    무르비2 리뷰   
1     REVIEW｜무릎 온열찜질기 &quot;메이크맨 무르비2&quot; 무릎찜질기 후기   20220129    무르비2 리뷰   
2                          메이크맨 무르비 2세대 무릎마사지기’ 솔직후기   20211125    무르비2 리뷰   
3                  원적외선 무릎마사지기 무르비2 무릎안마기 무릎찜질기 사용후기   20211009    무르비2 리뷰   
4                           무릎마사지와 무릎찜질이 가능한 무르비2 후기   20220120    무르비2 리뷰   
...                                              ...        ...        ...   
2520                 클럭 미니 마사지기 무릎 밴드 | 용도별 패드 체험 후기   20211030  무릎마사지기 리뷰   
2521                       웃챠 무릎마사지기 저주파마사지기 추천 솔직후기   20200324  무릎마사지기 리뷰   
2522              추석 선물세트로 준비해 본 효도 선물 클럭 무릎 마사지기 리뷰   20210831  무릎마사지기 리뷰   
2523                     [리뷰] 클럭(KLUG) 미니 마사지기 무릎 밴드   20200706  무릎마사지기 리뷰   
2524                                   클럭 무릎 마사지기 리뷰   20201112  무릎마사지기 리뷰   

     브랜드 명                                            content  content_cnt  \
0     메이크맨  저는 여름보다 유달리 겨울 추위를 잘 타는 체질이라 ... 히터가 작동되는 사무실에...         2435   
1     메이크맨  최근 들어 아부지가 무릎이 자주 아프다고 하셔서 집에서 편하게 통증을 줄일 수 있는...          872   
2     메이크맨  원적외선 무릎마사지기(온열/진동) SH-14K 76,900원 ➡️ 49,000원 요...         1893   
3     메이크맨  남편이 무릎이 안 좋아서 이것저것 무릎 보호대만  엄청 사서 썼는데 쓰던것 찾아보니...         1083   
4     메이크맨  저희 친정아빠가 술을 끊은지 몇년이 지났는데 작년 연말에 저희 남편이랑 갑자기 술을...         1188   
...    ...                                                ...          ...   
2520    클럭  얼마 전  클럭 미니 마사지기  발 패드, 대왕 패드, 큰 패드에  대한 소개를 간...         1389   
2521    클럭  안녕하세요 오늘 소개해드릴 제품은  #웃챠 #무릎마사지기 입니다!  제가 정말 다양...         1739   
2522    클럭  안녕하세요! 뷰스타 쪼에요 : ) 지난 8월은 눈코 뜰 새 없이 바쁜 한 달을 보냈...         2740   
2523    클럭  안녕하세요. 오늘은 클럭 미니 마사지기 무릎 밴드 구매 후기에요. 지난번 5월 가정...          572   
2524    클럭  무릎이 안 좋아서 샀는데  기대 이상입니다. 물리치료 한달 받아봤는데 저주파 치료할...          294   

      content_line                                           quote  ...  \
0               22        ['편리한 무선 온열찜질기진동 마사지 기능옥이 들어있어 원적외선 효과']  ...   
1               10                                              []  ...   
2               21                               ['•무릎건강 자가 테스트•']  ...   
3               23  ["'해당 포스팅은 업체에게 제품을 제공받아 주관적인 소견으로 작성하였습니다.'"]  ...   
4               16                                              []  ...   
...            ...                                             ...  ...   
2520            15                                              []  ...   
2521            20                                              []  ...   
2522            20                                              []  ...   
2523            12                     ['클럭 무릎 밴드2020.06.27', '끝']  ...   
2524             1                                              []  ...   

      ♥ 키워드 구매후기 키워드 한달 키워드  두달 키워드  세달 키워드  WEB_광고 분류1  WEB_광고 분류2  광고 분류1  \
0         0        0      0       0       0           0           0       1   
1         0        0      0       0       0           0           0       1   
2         0        0      0       0       0           0           0       1   
3         0        0      0       0       0           0           0       1   
4         0        0      0       0       0

## 3.4 json 파일을 읽어서 한 문장으로 만들기

In [70]:
# json 파일을 읽어서 OCR 결과를 하나의 텍스트 문장으로 만들기
def json_featuring(x):
    load_name  = x.values[0]
    try:
        with open(load_name, 'r', encoding='UTF-8') as json_file:
            json_data = json.load(json_file)
        result = " ".join(list(json_data.keys()))
    except FileNotFoundError:
        result = ""
    return result

In [71]:
df["context_img"] = df[["jsonFile"]].apply(lambda x: json_featuring(x), axis=1)

In [78]:
df[df["context_img"].str.contains("보스팅")]

index                                                url  \
1      1  https://blog.naver.com/PostView.naver?blogId=s...   

                                            title  post_date      검색어 브랜드 명  \
1  REVIEW｜무릎 온열찜질기 &quot;메이크맨 무르비2&quot; 무릎찜질기 후기   20220129  무르비2 리뷰  메이크맨   

                                             content  content_cnt  \
1  최근 들어 아부지가 무릎이 자주 아프다고 하셔서 집에서 편하게 통증을 줄일 수 있는...          872   

   content_line quote  ...  구매후기 키워드 한달 키워드 두달 키워드  세달 키워드  WEB_광고 분류1  \
1            10    []  ...         0      0      0       0           0   

   WEB_광고 분류2  광고 분류1  광고 분류2                    jsonFile  \
1           0       1       2  annotation/2/1_class2.json   

                                     context_img  
1  보스팅은 솔직하게 업체로부터 온 우 작성된 제공밭아 제품을 직접 체험 후기입니다.  

[1 rows x 53 columns]

In [81]:
# 정보 확인
for k in df["context_img"]:
    print(k)


보스팅은 솔직하게 업체로부터 온 우 작성된 제공밭아 제품을 직접 체험 후기입니다.


powe
a ce 칠옥타브
브랜드로부터 소정의 원고료를 이 작성되었습니다 지원받아 포스팅은 해당
beanpressi lon 행궁
1013 4.5점 a choice mare oo oro rome talv tazza travelers via 가능 결제 공식홈페이지 로마 리뷰 맛있는 목사 사진 시설 시용카드 이용 이지도 이탐리아 전체 주변 지역 커프 킴시0 트립어드바이저 평수
a the 구입 방을 사용한 수도 수수료를 실제 일환으로 있습니다. 제공 쿠팡파트너스 포스팅은 활동의 후 후기이자,
내림으
100x 검사 기는 기로 남음수 내부어 드리겠습니다 룸엄루이 모든 비프레소는 빈프레소는 사용하셔도 사용한 생기는 실시 아닐 안심하고 압습니다 에스티들 이슨니다 이하 있습니다 잔유물이오니 전수 전수검사(커프 전수검사를 전자동 제품 제품을 주조 진행하 책임 출고전 커피머신이 키스트 판마 패문어 하고 하기 하지 환불 환불제 휠니다 흐적(커피가루
andan pr

아이플르어 지사진

i테테크 각사한니다스 검색하고 꽃잔 네이버에서 늘러 늘러주시면 받아보실 배너를 빠르게 소식과 수 이벤트를 있어요 재미난 전문창작자 팬하기 팬하기틀
hing olid
beanpresso
th

oteat
beanpresso 블레드c 블렌드g 빈프레소 빈프레스 시그니처

anpresso pressi
11 1점 2점 3점 4점 5점
axcuve com macguyver naver 동영상 리뷰, 문의 제작
there thod
and c7
sorongersong
이사
so
d양 검색하고 네이버에서 리빙 전문들로거 팬하기
smisund

and helinox

an any as fre is kaviard nireditcol oj마켓 old rald resheasyeck sau thefl tot ure waw wwiv wwwastvecom 가기 눈 랭킹닭컴 마이셰프 무원 쇼핑하러 식품사 앙트레 오뚜기몰 정원e샵 직영몰 캐비아몰


httos instax liblo 결혼여행마집데이트 콩'sdaily
instax

instax minill rangi


10p 124,210 1개 단일상품 래덕발송 랜덤발송 로켓와우 미니11 블러쉬 세트 수량. 원 인스탁스 저용가빙 트나무데코 포토액자 필름 핑크 하 하드앨법
lew the 이마트
an instaxlen instox
mini11 nstax palette uceshoti
발생탈 사음 수노 수의이 슬식 이스니나 파트너스활동으로 후기이시만
instax instaxlensoo mini11

the
and
instax mini11

ange instax minil1
dicewhite instax mini1 mini11
tod whar
dean domave hito instaxminis mypalette 스
bo'so ind iz soit 리어 사라고 장채기이
and corot
insingers
10 instax retro the
instax

se

the
instax
319 arf doint le nstax print 넓어진 담다 당신의 더 사진으로 세상을

color pu 라임락퍼플 미니11 블러쉬핑크 스카이블루 아이스화이트 인스탁스 쟈골그레이 전용 케이스

instores
instay

and min nsta
텔나

ko
instax
dingesters


00
12.244 2020

0.31 instax lens ocus range

mini1 꿈꾸는 미물 여자 오늘도 이
공감라 김너다 림이 및국 은빛바다 정거장

a

jo sarker
instax
2편 그남 그래도 삿다 실수로 쓰자... 웰름을 흑백
3시 ol the thin u uin vew 가이트 금사사7 네이버쇼핑 다론 더보기 동영상 려모 미니즈 미스탁스 법적고지미안나 사이트 시식in 어학사산 이미지 이트 이프루어서 인스닥스미 인스닥스미니죽 인스락스미 인스탁스 인스탁스미 즉석 즉석필름 지도 지필름 트윈팩
23
to 라이
'업무식 망도론 소음(큰소리) 시간미 압주민의 조식 조심! 취침



In [114]:
# 정보 수정
df.loc[2221,"광고 분류1"] =1
df.loc[2221,"광고 분류2"] =1
df.loc[121,"광고 분류1"] =2
df.loc[121,"광고 분류2"] =2

# 4. 전처리된 데이터 저장

In [82]:
df.to_csv("./Total_Data2.csv",encoding="utf-8-sig",index=False)